scripts/deploy-on-kind.sh

```diff
@@ -193,8 +193,31 @@ function deploy_cf_k8s_controllers() {
     if [[ -n "${use_local_registry}" ]]; then
       make deploy-controllers-kind-local
     else
-      make deploy-controllers-kind
+      make deploy-controllers
     fi
+
+    if [[ "${OPENSSL_VERSION}" == "OpenSSL" ]]; then
+      openssl req -x509 -newkey rsa:4096 \
+        -keyout /tmp/app-tls.key \
+        -out /tmp/app-tls.crt \
+        -nodes \
+        -subj '/CN=*.vcap.me' \
+        -addext "subjectAltName = DNS:*.vcap.me" \
+        -days 365
+    else
+      openssl req -x509 -newkey rsa:4096 \
+        -keyout /tmp/api-tls.key \
+        -out /tmp/api-tls.crt \
+        -nodes \
+        -subj '/CN=*.vcap.me' \
+        -extensions SAN -config <(cat /etc/ssl/openssl.cnf <(printf "[ SAN ]\nsubjectAltName='DNS:*.vcap.me'")) \
+        -days 365
+    fi
+
+    kubectl create secret tls cf-k8s-workloads-ingress-cert \
+      -n cf-k8s-controllers-system \
+      --cert=/tmp/app-tls.crt \
+      --key=/tmp/app-tls.key
```

Comment:
Does it make sense to extract the secret generation to a separate script? We are already doing this for eirini: https://github.com/cloudfoundry/cf-k8s-controllers/blob/e2e-tests-run-on-non-kind-clusters/scripts/generate-eirini-certs-secret.sh

This would unclutter the script a little bit. Also given that these secret generation scripts are pretty similar maybe it is possible to somehow generalize and dry things up even further.

Reply:
Another thing to note is that `kubectl create secret` isn't idempotent, so we should either check to see if the secret already exists, or just delete any existing secret. Otherwise the deploy-on-kind.sh script will fail when we re-run it.

Reply:
My opinion is that `kubectl create` should never be used in scripts for this very reason. `kubectl apply -f <<EOF...` is a reasonable alternative.